_studio/shared/umc/codec/h264_dec/src/umc_h264_dec_slice_decoder_decode_pic.cpp

```diff
@@ -75,10 +75,15 @@ Status H264Slice::UpdateReferenceList(ViewList &views,
 
     VM_ASSERT(m_pCurrentFrame);
 
-    pRefPicList0 = m_pCurrentFrame->GetRefPicList(m_iNumber, 0)->m_RefPicList;
-    pRefPicList1 = m_pCurrentFrame->GetRefPicList(m_iNumber, 1)->m_RefPicList;
-    pFields0 = m_pCurrentFrame->GetRefPicList(m_iNumber, 0)->m_Flags;
-    pFields1 = m_pCurrentFrame->GetRefPicList(m_iNumber, 1)->m_Flags;
+    H264DecoderRefPicList* pH264DecRefPicList0 = m_pCurrentFrame->GetRefPicList(m_iNumber, 0);
+    H264DecoderRefPicList* pH264DecRefPicList1 = m_pCurrentFrame->GetRefPicList(m_iNumber, 1);
+    VM_ASSERT(pH264DecRefPicList0);
```

Comment:
emmm... You did not avoid deference actually. VM_ASSERT will be compiled out on release. Same is true for m_pCurrentFrame as well by the way.